sdk/textanalytics/ai-text-analytics/src/analyzeLro.ts

```diff
@@ -189,37 +189,35 @@ export function isAnalyzeDone(response: unknown): boolean {
 /**
  * @internal
  */
-export function processAnalyzeResult<TOptions extends OperationOptions>(
+export function processAnalyzeResult(
   // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
   client: GeneratedClient,
   documents: TextDocumentInput[],
-  options: TOptions
+  options: GeneratedClientAnalyzeStatusOptionalParams
 ): (result: unknown, state: AnalyzeActionsOperationState) => PagedAnalyzeActionsResult {
-  const pagedResult: PagedResult<
-    TOptions,
-    GeneratedClientAnalyzeStatusResponse,
-    AnalyzeActionsResult
-  > = {
-    sendGetRequest: (path: string, optionsParam: TOptions) =>
-      sendGetRequest(client, analyzeStatusOperationSpec, "AnalyzeStatus", optionsParam, path).then(
-        (response) => response.flatResponse as GeneratedClientAnalyzeStatusResponse
-      ),
-    buildPage: (flatResponse: GeneratedClientAnalyzeStatusResponse) => {
-      if (flatResponse) {
-        return createAnalyzeActionsResult(flatResponse, documents);
-      } else {
-        throw new Error("Analyze action has succeeded but there are no results!");
-      }
-    }
-  };
   return (_result: unknown, state: AnalyzeActionsOperationState): PagedAnalyzeActionsResult => {
     const pollingURL = (state as any).pollingURL;
-    const pagedIterator = getPagedAsyncIterator<
-      TOptions,
-      GeneratedClientAnalyzeStatusResponse,
-      AnalyzeActionsResult,
-      AnalyzeActionsResult
-    >(pagedResult, pollingURL, options);
+    const pagedResult: PagedResult<AnalyzeActionsResult> = {
+      link: pollingURL,
+      getPage: async (link: string, maxPageSize?: number) => {
+        const flatResponse = await sendGetRequest(
+          client,
+          analyzeStatusOperationSpec,
+          "AnalyzeStatus",
+          // if `top` is set to `undefined`, the default value will not be sent
+          // as part of the request.
+          maxPageSize ? { ...options, top: maxPageSize } : options,
+          link
+        ).then((response) => response.flatResponse as GeneratedClientAnalyzeStatusResponse);
```

Comment:
nit: feels bad to mix `then` and `await`

Reply:
Done.